docs/DEVELOP.md

```diff
@@ -1 +1,274 @@
+
+| Environment Variable         | Description     |
+|------------------------------|-----------------|
+| VERSION                      | Specifies the version of eraser to be used |
+| MANAGER_IMG                  | Defines the image url for the Eraser manager. Used for tagging, pulling and pushing the image |
+| ERASER_IMG                   | Defines the image url for the Eraser. Used for tagging, pulling and pushing the image |
+| COLLECTOR_IMG                | Defines the image url for the Collector. Used for tagging, pulling and pushing the image |
+
+### Linting
+
+- `make lint`
+
+    Lints the go code.
+
+    | Environment Variable         | Description     |
+    |------------------------------|-----------------|
+    | GOLANGCI_LINT                | Specifies the go linting binary to be used for linting. |
+
+### Development
+
+- `make generate` 
+    
+    Generates necessary files for the k8s api stored under `api/v1alpha1/zz_generated.deepcopy.go`. See the [kubebuilder docs](https://book.kubebuilder.io/cronjob-tutorial/other-api-files.html) for details.
+
+- `make manifests`
+
+    Generates the eraser deployment yaml files under `manifest_staging/deploy`.
+
+    Configuration Options:
+    | Environment Variable         | Description           |
+    |------------------------------|-----------------------|
+    | ERASER_IMG                   |  Defines the image url for the Eraser. |
+    | MANAGER_IMG                  | Defines the image url for the Eraser manager. |
+    | KUSTOMIZE_VERSION            | Define Kustomize version for generating manifests. |
+
+- `make test`
+
+    Runs the unit tests for the eraser project.
+
+    Configuration Options:
+    | Environment Variable         | Description           |
+    |------------------------------|-----------------------|
+    | ENVTEST                      | Specifies the envtest setup binary. |
+    | ENVTEST_K8S_VERSION          | Specifies the Kubernetes version for envtest setup command. |
+
+- `make e2e-test`
+
+    Runs e2e tests on a cluster.
+
+    Configuration Options:
+    | Environment Variable         | Description           |
+    |------------------------------|-----------------------|
+    | ERASER_IMG                   | Eraser image to be used for e2e test. |
+    | MANAGER_IMG                  | Eraser manager image to be used for e2e test. |
+    | KUBERNETES_VERSION           | Kubernetes version for e2e test. |
+    | TEST_COUNT                   | Sets repetition for test. Please refer to [go docs](https://pkg.go.dev/cmd/go#hdr-Testing_flags) for details. |
+    | TIMEOUT                      | Sets timeout for test. Please refer to [go docs](https://pkg.go.dev/cmd/go#hdr-Testing_flags) for details. |
+    | TESTFLAGS                    | Sets additional test flags |
+
+
+### Build
+
+- `make build`
+
+    Builds the eraser manager binaries.
+
+- `make run`
+
+    Runs the eraser manager on your local machine.
+
+- `docker-build-manager`
```

Comment:
for consistency, can we prefix with `make`? (or not prefix with `make`, either way)

Reply:
applies to the ones below as well